pkg/resource/google/google_bigquery_dataset_test.go

```diff
@@ -18,10 +18,15 @@ func TestAcc_Google_BigqueryDataset(t *testing.T) {
 		},
 		Checks: []acceptance.AccCheck{
 			{
-				// New resources are not visible immediately on GCP api after an apply operation
-				// Logic below retry driftctl scan until we can retrieve the results (infra will be in sync) and for maximum 60 seconds
+				// New resources are not visible immediately through GCP API after an apply operation.
+				// Logic below retries driftctl scan using a back-off strategy of retrying 'n' times
+				// and doubling the amount of time waited after each one.
 				ShouldRetry: func(result *test.ScanResult, retryDuration time.Duration, retryCount uint8) bool {
-					return !result.IsSync() && retryDuration < time.Minute
+					if result.IsSync() || retryDuration > 10*time.Minute {
+						return false
+					}
+					time.Sleep((2 * time.Duration(retryCount)) * time.Minute)
```

Comment:
Basing the exponential backoff on the retry count seems really weird to me? You mentioned earlier you thought exponential backoff might not be optimal, and FWIW I'm totally fine with that 😁 

Also every time I said linear backoff earlier, I actually meant constant backoff - sorry!

WDYT of a simple hardcoded constant backoff to start with, and we can re-evaluate based on the flake rate after that?

```suggestion
					time.Sleep(10 * time.Second)
```

Reply:
Even if we do keep the backoff as a multiple of retryCount as you originally suggest, `count * 2m` seems a bit long for the initial wait. WDYT?

Reply:
> You mentioned earlier you thought exponential backoff might not be optimal, and FWIW I'm totally fine with that grin

I did but I also think exponential backoff can be good with a reasonable time limit. Your initial suggestion was good actually! A constant backoff can lead to more friction as it's impossible to know what's the right time frame to choose in this case. Exponential suggests we keep using a best-effort approach : retrying immediately, then after just 2 minutes, then after a bit longer... 

If after all we stay on a hardcoded backoff, I think 3 minutes per retry for a maximum of 3 attempts (9min) is enough for now. WDYT?

Reply:
Yeah let's start with this actually, I'll approve and we can always re-evaluate if the flakes continue.

FWIW I think this is linear backoff, not exponential (2,4,6.., not 2,4,8,16...), but whatever 😉